File: pump/equil/src/main/java/app/aaps/pump/equil/manager/command/CmdDevicesGet.java
package app.aaps.pump.equil.manager.command;


import app.aaps.core.interfaces.logging.LTag;
import app.aaps.pump.equil.database.EquilHistoryRecord;
import app.aaps.pump.equil.manager.Utils;


public class CmdDevicesGet extends BaseSetting {
    public CmdDevicesGet() {
        super(System.currentTimeMillis());
        port = "0000";
    }

    @Override
    public byte[] getFirstData() {
        byte[] indexByte = Utils.intToBytes(pumpReqIndex);
        byte[] data2 = new byte[]{0x02, 0x00};
        byte[] data = Utils.concat(indexByte, data2);
        pumpReqIndex++;
        return data;
    }

    public byte[] getNextData() {
        byte[] indexByte = Utils.intToBytes(pumpReqIndex);
        byte[] data2 = new byte[]{0x00, 0x00, 0x01};
        byte[] data = Utils.concat(indexByte, data2);
        pumpReqIndex++;
        return data;
    }

    @Override
    public void decodeConfirmData(byte[] data) {
        int value = Utils.bytesToInt(data[7], data[6]);

        String firmwareVersion = data[18] + "." + data[19];
        aapsLogger.debug(LTag.PUMPCOMM, "CmdGetDevices====" +
                Utils.bytesToHex(data) + "=====" + value + "===" + firmwareVersion);
        equilManager.setFirmwareVersion(firmwareVersion);
        synchronized (this) {
            setCmdStatus(true);
            notify();
        }
    }

    @Override public EquilHistoryRecord.EventType getEventType() {
        return EquilHistoryRecord.EventType.READ_DEVICES;
    }
}

File: pump/equil/src/main/kotlin/app/aaps/pump/equil/manager/command/CmdDevicesGet.kt
package app.aaps.pump.equil.manager.command

import app.aaps.core.interfaces.logging.AAPSLogger
import app.aaps.core.interfaces.logging.LTag
import app.aaps.core.keys.interfaces.Preferences
import app.aaps.pump.equil.database.EquilHistoryRecord
import app.aaps.pump.equil.manager.EquilManager
import app.aaps.pump.equil.manager.Utils

class CmdDevicesGet(
    aapsLogger: AAPSLogger,
    preferences: Preferences,
    equilManager: EquilManager
) : BaseSetting(System.currentTimeMillis(), aapsLogger, preferences, equilManager) {

    init {
        port = "0000"
    }

    override fun getFirstData(): ByteArray {
        val indexByte = Utils.intToBytes(pumpReqIndex)
        val data2 = byteArrayOf(0x02, 0x00)
        val data = Utils.concat(indexByte, data2)
        pumpReqIndex++
        return data
    }

    override fun getNextData(): ByteArray {
        val indexByte = Utils.intToBytes(pumpReqIndex)
        val data2 = byteArrayOf(0x00, 0x00, 0x01)
        val data = Utils.concat(indexByte, data2)
        pumpReqIndex++
        return data
    }

    override fun decodeConfirmData(data: ByteArray) {
        val value = Utils.bytesToInt(data[7], data[6])

        val firmwareVersion = data[18].toString() + "." + data[19]
        aapsLogger.debug(
            LTag.PUMPCOMM, "CmdGetDevices====" +
                Utils.bytesToHex(data) + "=====" + value + "===" + firmwareVersion
        )
        equilManager.setFirmwareVersion(firmwareVersion)
        synchronized(this) {
            cmdStatus = true
            (this as Object).notify()
        }
    }

    override fun getEventType(): EquilHistoryRecord.EventType? = EquilHistoryRecord.EventType.READ_DEVICES
}
